WalletWasabi.Backend/Filters/IdempotentActionFilterAttributeImpl.cs

```diff
@@ -60,7 +60,7 @@ public override async Task OnActionExecutionAsync(ActionExecutingContext context
 				{
 					try
 					{
-						context.Result = await cachedResponseTcs!.Task.WithAwaitCancellationAsync(RequestTimeout).ConfigureAwait(false);
+						context.Result = await cachedResponseTcs!.Task.WaitAsync(RequestTimeout).ConfigureAwait(false);
```

Comment:
Is this not changing the behavior of having `TimeoutException` instead of `OperationCanceledException`?

Reply:
Yes, unfortunately. Even though it is somewhat good to distinct between: Something was cancelled and something timed out. Those can be very different things.

Anyway, there are several ways we can approach this (if we want to make the switch from `WithAwaitCancellationAsync` to `WaitAsync` at all). The ideas are sorted by "the level of behavior preserving":

1. Preserve `WithAwaitCancellationAsync` but implement it with something like:
    ```cs
    try { return await task.WaitAsync(..)[...]; } catch (TimeoutException e) { throw new OperationCancelledException(); }`
    ```
2. Identify where it is necessary to add catching of `TimeoutException` case by case based on changes in this PR.
3. Change `catch` clauses that catch `OperationCancelledException` to also catch `TimeoutException` also everywhere.
4. Do nothing and add `catch` clauses as needed.

It's worth noting that mostly we catch `OperationCanceledException` handle that with different log level than other exceptions.

Reply:
I would prefer to avoid breaking changes and go with option nr. 1 first.

Reply:
Yes, makes sense. Changed.

Reply:
ACK